test/comments.d

```diff
@@ -0,0 +1,243 @@
+import app;
+import utils;
+
+import vibe.d;
+import std.functional;
+import std.stdio;
+
+// existing dlang bot comment -> update comment
+unittest
+{
+    auto expectedURLs = ["/github/repos/dlang/phobos/pulls/4921/commits",
+                         "/github/repos/dlang/phobos/issues/4921/labels",
+                         "/github/repos/dlang/phobos/issues/4921/comments",
+                         "/bugzilla/buglist.cgi?bug_id=8573&ctype=csv&columnlist=short_desc",
+                         "/github/repos/dlang/phobos/issues/comments/262784442"];
+
+    payloader = (scope HTTPServerRequest req, scope HTTPServerResponse res) {
+        if (req.requestURL == "/github/repos/dlang/phobos/issues/comments/262784442")
+        {
+            assert(req.method == HTTPMethod.PATCH);
+            auto expectedComment =
+`Fix | Bugzilla | Description
+--- | --- | ---
+✗ | [8573](%s/show_bug.cgi?id=8573) | A simpler Phobos function that returns the index of the mix or max item
+`.format(bugzillaURL);
+            assert(req.json["body"].get!string == expectedComment);
+            res.writeVoidBody;
+            return DirectionTypes.STOP;
+        }
+        return DirectionTypes.CONTINUE;
+    }.toDelegate();
+
+    "./payloads/github_hooks/dlang_phobos_synchronize_4921.json".buildGitHubRequest(expectedURLs);
+}
+
+// no existing dlang bot comment -> create comment
+unittest
+{
+    auto expectedURLs = ["/github/repos/dlang/phobos/pulls/4921/commits",
```

Comment:
Looks a bit as if it could be done somewhat similar to the router.
```d
setAPIExpectations!(
    "/github/repos/dlang/phobos/pulls/4921/commits",
    "/github/repos/dlang/phobos/issues/4921/labels",
    //...
    "/github/repos/dlang/phobos/issues/4921/comments", (scope req, scope res) {
        // assert(req == );
    });
postStatus("...");
```